rs/bitcoin/ckbtc/minter/src/lib.rs

```diff
@@ -1024,25 +1015,15 @@ pub fn build_unsigned_transaction(
 
     let amount = outputs.iter().map(|(_, amount)| amount).sum::<u64>();
 
-    let input_utxos = utxos_selection(amount, minter_utxos, outputs.len());
-
     if input_utxos.is_empty() {
         return Err(BuildTxError::NotEnoughFunds);
     }
 
-    // This guard returns the selected UTXOs back to the available_utxos set if
-    // we fail to build the transaction.
-    let utxos_guard = guard(input_utxos, |utxos| {
-        for utxo in utxos {
-            minter_utxos.insert(utxo);
-        }
-    });
```

Comment:
Please correct me if I am wrong, but I believe this guard was ineffective, because there is no await between its creation and its diffusion (at the end of this function).